client/lib/Agent.ts

```diff
@@ -329,6 +329,13 @@ export default class Agent extends AwaitedEventTarget<{ close: void }> {
       return onrejected(err);
     }
   }
+
+  public toJSON(): any {
+    // return empty so we can
+    return {
+      type: 'Agent',
```

Comment:
Why not use `this.constructor.name`?

Reply:
There's not a base class. Are you thinking it would just be better if someone copies and pastes a whole class?

Reply:
It's just a safety precaution in case the class gets renamed. The string her is unlikely to be caught.

Reply:
Ok, fine by me. You want to just update when you pull it in?

Reply:
FWIW, I considered making this more of an instructional message - like, toJSON in SecretAgent doesn't attempt to serialize all properties since they are async